llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -85,6 +85,64 @@ void emitBundlesInScoreboardDelta(
   }
 }
 
+ResourceScoreboard<FuncUnitWrapper>
+createBottomUpScoreboard(ArrayRef<MachineBundle> Bundles,
+                         const AIEHazardRecognizer &HR) {
+  const unsigned NumBundles = Bundles.size();
+  const unsigned RequiredCycles = HR.getConflictHorizon();
+
+  ResourceScoreboard<FuncUnitWrapper> Scoreboard;
+  Scoreboard.reset(HR.getMaxLookAhead());
+
+  // We have less known bundles than the minimum number required for
+  // correctness. Conservatively block some cycles.
+  if (NumBundles < RequiredCycles) {
+    unsigned CyclesToBlock = RequiredCycles - Bundles.size();
+    for (unsigned Num = 0; Num < CyclesToBlock; ++Num) {
+      Scoreboard[0].blockResources();
+      Scoreboard.recede();
+    }
+  }
+
+  // Do not emit more Bundles than required for correctness.
+  ArrayRef<MachineBundle> MinBundles(
+      Bundles.begin(), Bundles.begin() + std::min(NumBundles, RequiredCycles));
+  for (const MachineBundle &B : reverse(MinBundles)) {
+    for (MachineInstr *MI : B.getInstrs())
+      HR.emitInScoreboard(Scoreboard, MI->getDesc(), HR.getMemoryBanks(MI), 0);
+    Scoreboard.recede();
+  }
+  return Scoreboard;
+}
+
+/// Replay the bundles in \p SuccBundles bottom-up, followed by those in
+/// \p PredBundles. If that causes a resource conflict, return the instruction
```

Comment:
I need to adjust the comment into something like `Replay the \p PredBundles bottom-up into \p ScoreBoard.`